tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/EndpointDetector.cs

```diff
@@ -0,0 +1,296 @@
+// <copyright file="EndpointDetector.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Datadog.Trace.Pdb;
+using Datadog.Trace.VendoredMicrosoftCode.System.Collections.Immutable;
+using Datadog.Trace.VendoredMicrosoftCode.System.Reflection.Metadata;
+using Datadog.Trace.VendoredMicrosoftCode.System.Reflection.Metadata.Ecma335;
+
+namespace Datadog.Trace.Debugger.SpanCodeOrigin;
+
+internal static class EndpointDetector
+{
+    private static readonly HashSet<string> ControllerAttributes =
+    [
+        "Microsoft.AspNetCore.Mvc.ApiControllerAttribute",
+        "Microsoft.AspNetCore.Mvc.ControllerAttribute",
+        "Microsoft.AspNetCore.Mvc.RouteAttribute"
+    ];
+
+    private static readonly HashSet<string> ControllerBaseNames =
+    [
+        "Microsoft.AspNetCore.Mvc.Controller",
+        "Microsoft.AspNetCore.Mvc.ControllerBase"
+    ];
+
+    private static readonly HashSet<string> ActionAttributes =
```

Comment:
There's also [the `HttpMethodAttribute`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.routing.httpmethodattribute?view=aspnetcore-9.0) which accepts a collection of verbs

Reply:
Fixed in a5a2c085c57f23a246a2b781dec1bc9354502a16